File: apps/api/src/shared/domain/csv.ts
/** Quote cells without discarding text, and neutralize spreadsheet formulas. */
export function csvCell(value: unknown): string {
  const text = String(value ?? "");
  const safe =
    /^[\s\u0000-\u001f]*[=+@-]/u.test(text) || /^[\t\r\n]/u.test(text)
      ? `'${text}`
      : text;
  return `"${safe.replace(/"/g, '""')}"`;
}

File: apps/mobile/lib/domain/models/csv.dart
/// Preserve exported text and prevent spreadsheet programs interpreting formulas.
String csvCell(Object? value) {
  final text = (value ?? '').toString();
  final formula =
      RegExp(r'^[\s\u0000-\u001f]*[=+@-]').hasMatch(text) ||
      RegExp(r'^[\t\r\n]').hasMatch(text);
  final safe = formula ? "'$text" : text;
  return '"${safe.replaceAll('"', '""')}"';
}
